clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemFactory.java

```diff
@@ -153,22 +146,52 @@ protected SystemProducer createSystemProducer(
       Optional<SSLFactory> sslFactory,
       Optional<String> partitioners) {
     return new VeniceSystemProducer(
-        veniceD2ZKHost,
-        veniceD2Service,
+        primaryControllerColoD2ZKHost,
+        primaryControllerColoD2ZKHost,
+        primaryControllerD2Service,
         storeName,
         venicePushType,
         samzaJobId,
         runningFabric,
-        config.getBoolean(VALIDATE_VENICE_INTERNAL_SCHEMA_VERSION, false),
+        config.getBoolean(VALIDATE_VENICE_INTERNAL_SCHEMA_VERSION, true),
         this,
         sslFactory,
         partitioners);
   }
 
-  // Extra `Config` parameter is to ease the internal implementation
+  /**
+   * @deprecated Left in to maintain backward compatibility
+    */
+  @Deprecated
   protected SystemProducer createSystemProducer(
-      String veniceD2ZKHost,
-      String veniceD2Service,
+      String primaryControllerColoD2ZKHost,
+      String primaryControllerD2Service,
+      String storeName,
+      Version.PushType venicePushType,
+      String samzaJobId,
+      String runningFabric,
+      boolean verifyLatestProtocolPresent,
+      Config config,
+      Optional<SSLFactory> sslFactory,
+      Optional<String> partitioners) {
+    return createSystemProducer(
+        primaryControllerColoD2ZKHost,
+        primaryControllerColoD2ZKHost,
+        primaryControllerD2Service,
+        storeName,
+        venicePushType,
+        samzaJobId,
+        runningFabric,
+        verifyLatestProtocolPresent,
+        config,
+        sslFactory,
+        partitioners);
+  }
+
+  protected SystemProducer createSystemProducer(
+      String veniceChildD2ZkHost,
+      String primaryControllerColoD2ZKHost,
```

Comment:
Can we add a javaDoc to explain the differences of these two parameters: `veniceChildD2ZkHost` and `primaryControllerColoD2ZKHost`; it will be helpful to point out when they will be the same and when they are different.

Reply:
Sure. Added